Intermittent failure in testMultithreadedNestedSpansAsync

This is due to the same random number being generated in testMultithreadedNestedSpansAsync. Instead, we should use some monotonically increasing number.

Details:

```
java.lang.AssertionError: expected [true] but found [false]
        at org.testng.Assert.fail(Assert.java:94)
        at org.testng.Assert.failNotEquals(Assert.java:496)
        at org.testng.Assert.assertTrue(Assert.java:42)
        at org.testng.Assert.assertTrue(Assert.java:52)
        at org.eclipse.microprofile.opentracing.tck.OpentracingClientTests.runNestedTests(OpentracingClientTests.java:639)
        at org.eclipse.microprofile.opentracing.tck.OpentracingClientTests.testMultithreadedNestedSpansAsync(OpentracingClientTests.java:557)
```

Failure in the TCK here:

```
            // If this assertion fails, it means that the unique ID
            // in the root span URL doesn't match any of the
            // unique IDs that we sent in the requests above.
            boolean removeResult = uniqueIds.remove(uniqueId);

            if (!removeResult) {
                debug("Unique ID " + uniqueId + " not found in request list. Span: " + rootSpan);
            }
```

This testMultithreadedNestedSpansAsync test executes 100 concurrent HTTP requests with a unique ID in the URL of each one. However, we can see that the ID 242096 has been generated twice:

            [...]
            <line>
              <![CDATA[Executing http://localhost:8010/opentracing/rest/testServices/nested?nestDepth=1&nestBreadth=2&async=true&data=242096&failNest=false]]>
            </line>
            [...]
            <line>
              <![CDATA[Executing http://localhost:8010/opentracing/rest/testServices/nested?nestDepth=1&nestBreadth=2&async=true&data=600395&failNest=false]]>
            </line>
            <line>
              <![CDATA[Executing http://localhost:8010/opentracing/rest/testServices/nested?nestDepth=1&nestBreadth=2&async=true&data=242096&failNest=false]]>
            </line>
            [...]

When the ID is generated, it is put into a Set, meaning that multiple of the same IDs will only be put once. Later, when the test is going through the root spans and confirming the right IDs are there, it removes the ID from the set, and so when it's verifying the second of these requests, that ID is no longer in the set and the assertion fails.

So the issue is that the random number generator returned the same number twice within 100 calls:

```
    /**
     * Get a random integer.
     * @return Random integer.
     */
    private int getRandomNumber() {
        int uniqueId = ThreadLocalRandom.current().nextInt(1, 999999);
        return uniqueId;
    }
```

The fix is to change from a random number generator to a synchronized, monotonically increasing number generator: https://github.com/eclipse/microprofile-opentracing/blob/1.0-branch/tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpentracingClientTests.java#L1093

--- a/tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpentracingClientTests.java
+++ b/tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpentracingClientTests.java
@@ -32,9 +32,9 @@ import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
-import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.TimeUnit;
 
+import java.util.concurrent.atomic.AtomicInteger;
 import javax.ws.rs.HttpMethod;
 import javax.ws.rs.client.Client;
 import javax.ws.rs.client.ClientBuilder;
@@ -76,6 +76,7 @@ import io.opentracing.tag.Tags;
  */
 public class OpentracingClientTests extends Arquillian {
     private static final String JAXRS_COMPONENT = "jaxrs";
+    private final AtomicInteger idCounter = new AtomicInteger(0);
 
     /** Server app URL for the client tests. */
     @ArquillianResource
@@ -1104,7 +1105,6 @@ public class OpentracingClientTests extends Arquillian {
      * @return Random integer.
      */
     private int getRandomNumber() {
-        int uniqueId = ThreadLocalRandom.current().nextInt(1, 999999);
-        return uniqueId;
+        return idCounter.incrementAndGet();
     }
 }

